### ISSUE TITLE
Attaching additional security group to EKS nodes not working

### ISSUE THREAD

**Issue by aaron-hastings-travelport** [terraform-aws-modules/terraform-aws-eks#1846]
## Description

Hi. We are required to open SSH on our EKS-managed nodes and allow connections from a specific private IP CIDR range (the "SSH client instances" live in another AWS account). The relevant VPCs are peered using Transit Gateway and I can verify that network connectivity between the clients and EKS nodes works correctly.

We are using v17.24.0 of the module, as we are not yet able to use v18.

I have manually created an SSH key pair and security group. The security group allows ingress of all traffic from the aforementioned CIDR range.

In the module's `node_group_defaults` block, I have added the following:

```
node_group_defaults = {
    [...]
    key_name                  = "my-ssh-key"  # Key created outside of Terraform (already exists)
    source_security_group_ids = ["sg-123456789abcdef"]  # SG created outside of Terraform (already exists)
    [...]
}
```

...but this is not working as I'd have expected. SSH is indeed enabled on the EKS nodes, but the additional security group is not attached to the worker nodes.

Have I misunderstood something in this approach? Is there an alternative means of simply "Letting this one extra CIDR block access my EKS nodes"?

Thanks - Aaron

## Versions
 - Terraform:
   ```
   Terraform v1.0.10
   on darwin_amd64
   ```

 - Provider(s):
   ```
   Terraform v1.0.10
   on darwin_amd64
   + provider registry.terraform.io/hashicorp/aws v3.74.0
   + provider registry.terraform.io/hashicorp/cloudinit v2.2.0
   + provider registry.terraform.io/hashicorp/kubernetes v2.7.1
   + provider registry.terraform.io/hashicorp/local v2.1.0
   + provider registry.terraform.io/terraform-aws-modules/http v2.4.1
   ```

 - Module: `eks`

### Code Snippet to Reproduce
```
module "eks" {
  source  = "terraform-aws-modules/eks/aws"
  version = "17.24.0"

  cluster_name              = var.cluster_name
  cluster_version           = var.cluster_version
  cluster_enabled_log_types = ["api", "audit", "authenticator", "controllerManager", "scheduler"]

  vpc_id  = var.vpc_id
  subnets = var.subnets

  tags = var.tags

  write_kubeconfig = false
  enable_irsa      = true
  manage_aws_auth  = true

  map_roles = local.map_roles_aws_auth

  cluster_encryption_config = [
    {
      provider_key_arn = aws_kms_key.eks.arn
      resources        = ["secrets"]
    }
  ]

  # IAM policy will be attached to vpc-cni add-on only
  attach_worker_cni_policy = false

  cluster_endpoint_public_access       = false
  cluster_endpoint_public_access_cidrs = false

  cluster_endpoint_private_access                = true
  cluster_create_endpoint_private_access_sg_rule = true
  cluster_endpoint_private_access_cidrs          = var.cluster_endpoint_private_access_cidrs
  cluster_endpoint_private_access_sg             = var.cluster_endpoint_private_access_sg

  # Timeout after 10 minutes
  wait_for_cluster_timeout = 600

  node_groups_defaults = {
    desired_capacity          = 2
    min_capacity              = 2
    max_capacity              = 2
    instance_types            = ["m5a.xlarge"]
    create_launch_template    = true
    disk_type                 = "gp3"
    disk_size                 = 100
    key_name                  = "my-ssh-key"  # Key created outside of Terraform (already exists)
    source_security_group_ids = ["sg-123456789abcdef"]  # SG created outside of Terraform (already exists)
  }

  node_groups = var.node_groups
}
```

## Expected behavior

- SSH is enabled for the EKS-managed nodes with specified key pair.
- Additional security group specified in `source_security_group_ids` is attached to the nodes which permits access from specified private CIDR range.

## Actual behavior

- SSH is indeed enabled for the EKS-managed nodes with specified key pair (all good here!).
- Only default security group is attached to the EKS-managed nodes. Security group referenced in `source_security_group_ids` is ignored (Terraform says "no changes").

**Comment by auyer** [comment]
Use `vpc_security_group_ids` instead of  `source_security_group_ids`.
Worked for me !

**Comment by bryantbiggs** [comment]
`source_security_group_ids` is not a valid variable - as @auyer pointed out, you should use the variable `vpc_security_group_ids` - please refer to examples for usage patterns
- https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/examples/complete/main.tf#L82
- https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/examples/complete/main.tf#L114

**Comment by aaron-hastings-travelport** [comment]
Just to confirm, is `vpc_security_group_ids` available in v17.24.0? In the end I discovered the `worker_additional_security_group_ids` variable which worked for me.
